indico/modules/events/registration/controllers/api/checkin.py

```diff
@@ -130,3 +133,33 @@ def _process_args(self):
 
     def _process_GET(self):
         return CheckinRegistrationSchema().jsonify(self.registration)
+
+
+class RHCheckinAPIRegistrationCustomQRCode(RHCheckinAPIBase):
+    """Get indico style ticket details for a specific registration from a custom plugin QR Code."""
+
+    def _process_args(self):
+        pass
+
+    def _check_access(self):
+        pass
+
+    @use_kwargs({
+        'data': fields.String(required=True),
+        'name': fields.String(required=True),
+    }, location='json')
+    def _process_POST(self, data, name):
+        sig_rvs = values_from_signal(
+            signals.event.registration.handle_custom_ticket_qr_code.send(data, qr_code_name=name),
```

Comment:
I'd rather make the name the `sender` since plugins can easily register a signal handler only for a particular sender, and pass the data as a kwarg instead.